src/Mapping/OwningSideMapping.php

```diff
@@ -11,6 +11,8 @@ abstract class OwningSideMapping extends AssociationMapping
      * The name of the field that completes the bidirectional association on
      * the inverse side. This key must be specified on the owning side of a
      * bidirectional association.
+     *
+     * @var non-falsy-string|null
```

Comment:
The point of this PR was to eliminate the implicit casts of this property to a boolean, right? in that case, I don't understand why we would still need to point out that this string is not falsy.

Reply:
We don't _need_ to, it's more something that is derived from the comment above it, and that makes the rest of the PR OK. Because it's a non falsy string, we don't need to replace the implicit cast with strict comparisons with `''` or `'0'`.

Reply:
Removed it since it serves now purpose besides making this PR slightly easier to understand.